File: src/main/java/com/leon/bugreport/API/CacheCleanupListener.java
package com.leon.bugreport.API;

import org.bukkit.scheduler.BukkitRunnable;

import static com.leon.bugreport.BugReportManager.plugin;

public class CacheCleanupListener {
	private static final int TICKS_PER_SECOND = 20;
	private static final int SECONDS_PER_MINUTE = 60;
	private static final int MINUTES_PER_HOUR = 60;
	private static final long TWO_HOUR_TICKS = TICKS_PER_SECOND * SECONDS_PER_MINUTE * MINUTES_PER_HOUR * 2; // 2 hours in ticks

	public CacheCleanupListener() {
		new BukkitRunnable() {
			@Override
			public void run() {
				DataSource.cleanOutdatedCache(false);
			}
		}.runTaskTimerAsynchronously(plugin, 0L, TWO_HOUR_TICKS); // Every 2 hours
	}
}

File: src/main/java/com/bugreportmc/bugreport/api/CacheCleanupListener.kt
package com.bugreportmc.bugreport.api

import com.bugreportmc.bugreport.BugReportManager
import com.bugreportmc.bugreport.BugReportPlugin.Companion.plugin
import com.bugreportmc.bugreport.api.DataSource.cleanOutdatedCache
import org.bukkit.scheduler.BukkitRunnable

class CacheCleanupListener {
	init {
		object : BukkitRunnable() {
			override fun run() {
				cleanOutdatedCache(false)
			}
		}.runTaskTimerAsynchronously(plugin, 0L, TWO_HOUR_TICKS)
	}

	companion object {
		private const val TICKS_PER_SECOND = 20
		private const val SECONDS_PER_MINUTE = 60
		private const val MINUTES_PER_HOUR = 60
		private const val TWO_HOUR_TICKS = (TICKS_PER_SECOND * SECONDS_PER_MINUTE * MINUTES_PER_HOUR * 2).toLong()
	}
}
